guide/students/student_management_service.bal

```diff
@@ -0,0 +1,324 @@
+        log:printError("error test");
+        response.setTextPayload("Test Error made");
+        _ = httpConnection->respond(response);
+    }
+
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "/deleteStu/{stuId}"
+    }
+    // Delete Students resource for deleteing a student using id.
+    deleteStudent(endpoint httpConnection, http:Request request, int stuId) {
+        requestCounts++;
+        http:Response response;
+        json status = {};
+
+        // Calling deleteData function with id as parameter and get a return json object.
+        var ret = deleteData(stuId);
+        io:println(ret);
+
+        // Pass the obtained json object to the request.
+        response.setJsonPayload(ret);
+        _ = httpConnection->respond(response);
+        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
+        _ = observe:addTagToSpan(spanId = -1, "tot_requests", <string>requestCounts);
+        _ = observe:addTagToSpan(spanId = -1, "error_counts", <string>errors);
+    }
+
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "/getMarks/{stuId}"
+    }
+    // Get marks resource for obtaining marks of a particular student.
+    getMarks(endpoint httpConnection, http:Request request, int stuId) {
+        requestCounts++;
+        http:Response response;
+        json result;
+
+        // Self defined span for observability purposes.
+        int firstsp = check observe:startSpan("First span");
+        // Request made for obtaining marks of the student with the respective stuId to marks Service.
+        var requ = marksService->get("/marks/getMarks/" + untaint stuId);
+
+        match requ {
+            http:Response response2 => {
+                var msg = response2.getJsonPayload();
+                // Gets the Json object.
+                match msg {
+                    json js => {
+                        result = js;
+                    }
+
+                    error er => {
+                        log:printError(er.message, err = er);
+                    }
+                }
+            }
+            error err => {
+                log:printError(err.message, err = err);
+            }
+        }
+        // Stopping the previously started span.
+        _ = observe:finishSpan(firstsp);
+        //Sending the Json to the client.
+        response.setJsonPayload(untaint result);
+        _ = httpConnection->respond(response);
+
+        //  The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
+        _ = observe:addTagToSpan("tot_requests", <string>requestCounts);
+        _ = observe:addTagToSpan("error_counts", <string>errors);
+    }
+}
+
+// Function to insert values to database
+  # `insertData()` is a function to add data to student records database.
+  # + name - This is the name of the student to be added.
```

Comment:
Shall we add a empty `#` between description and the parameters? Please fix in all places.